experiments/templatetags/experiments.py

```diff
@@ -85,26 +111,263 @@ def experiment(parser, token):
+                continue  # just ignore commas
+            # {% experiment %} tag only accepts strings, i.e. Const:
+            args.append(nodes.Const(parser.stream.current.value))
+            next(parser.stream)
+
+        # verify tag syntax:
+        tokens = [nodes.Const('experiment')] + args
+        try:
+            _parse_token_contents(list(map(attrgetter('value'), tokens)))
+        except ValueError:
+            raise TemplateSyntaxError(
+                "Syntax should be like: "
+                "{% experiment experiment_name"
+                " alternative [weight=val] [user=val] %}",
+                lineno,
+            )
+
+        # fill in default values:
+        while len(args) < 4:
+            args.append(nodes.Const(None))
+
+        # additional args:
+        args.append(nodes.ContextReference())
+
+        # parse the body of the block up to `endexperiment` and
+        # drop the needle (which will always be `endexperiment`):
+        body = parser.parse_statements(
+            ['name:endexperiment'], drop_needle=True)
+
+        # Jinja2 callbacky nodey magic:
+        call_node = self.call_method('render_experiment', args, lineno=lineno)
+        return nodes.CallBlock(call_node, [], [], body).set_lineno(lineno)
+
+    def render_experiment(
+            self, experiment_name, alternative, weight, user_variable,
+            context, caller):
+        """Callback to render {% experiment ... %} tags"""
+
+        experiment = experiment_manager.get_experiment(experiment_name)
+        if experiment:
+            # create alternative on the fly (write it to DB) if not existing:
+            experiment.ensure_alternative_exists(alternative, weight)
+
+        # Get User object
+        if user_variable:
+            auth_user = context[user_variable]
+            user = participant(user=auth_user)
+        else:
+            request = context.get('request')
+            user = participant(request)
+
+        # Should we render?
+        if user.is_enrolled(experiment_name, alternative):
+            return caller()
+        else:
+            return nodes.Markup()  # empty node
+
+    def parse_experiments_confirm_human(self, parser):
+        """Parse {% experiments_confirm_human %} tags"""
+        lineno = parser.stream.current.lineno
+        args = [nodes.ContextReference()]
+        node = self.call_method(
+            'render_experiments_confirm_human', args, lineno=lineno)
+        return nodes.CallBlock(node, [], [], []).set_lineno(lineno)
+
+    def render_experiments_confirm_human(self, context, caller):
+        """Callback to render {% experiments_confirm_human %} tags"""
+        tmplt = template.loader.get_template('experiments/confirm_human.html')
+        context_dict = dict(context)
+        context_dict.update(
+            _experiments_confirm_human(context_dict)
+        )
+        return tmplt.render(context_dict)
+
+    def parse_experiment_goal(self, parser):
```

Comment:
another cosmetic, can we join all the functions with the form "parse_{}" so we can know which possibilities are available?

Reply:
I mean, align them like:
```
def parse_opt1(params...):
    ....

def parse_opt2(params):
    .....